### ISSUE TITLE
Pause reason is not shown for multi session debugging, or vscode-pwa

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#85324]
Noticed that when debugging a simple node script with vscode-pwa, I don't get the pause reason (like "paused on debugger statement") which normally shows up in the Call Stack header.

![image](https://user-images.githubusercontent.com/323878/69366984-3eddea00-0c4c-11ea-80d5-952e583eaffa.png)

I guess this is because we enter the multi thread debugging mode. I am wondering whether

- We should still show the reason when in multi thread debugging, in the same place, based on the selected thread. I think this is useful info
- Or whether we should collapse the tree when only debugging one thread, which is still the most common case for node debugging, and only show an actual tree when attaching to another process. I think this could add some confusing overhead for people debugging node.

FYI @weinand @connor4312

**Comment by isidorn** [comment]
Hey Rob,

yes you are right, we do not show the data in the call stack header since there are multiple programs. And instead of that we show it inline. In the picture you see it as `PAUSED` next to `[98451]` - which I believe is a sub session. If this is a thread, let me know since that shows a bug - Launch Program should also have `RUNNING`.
Instead of `PAUSED` we could say `PAUSED ON ${REASON}`, however that might take up to much space and that is the reason why I did not originaly do this.

And yes we can introduce the label in the title area based on focus, however I think that would introduce too much duplication (since we already show this data inline).

I am open for suggestions

As for collapsing the tree when only having one thread, we had varioas discussions on this topic and decided on the current solution.  Similar issue https://github.com/microsoft/vscode/issues/84601

**Comment by roblourens** [comment]
Showing it inline might be too wordy or it might be ok. Showing it in the title bar might be ok too. But I think we shouldn't get rid of that info, because issues have shown that users sometimes honestly don't understand why they paused, and any extra hint is helpful.

What you said in that issue about collapsing the tree,

> start two debugging sessions, stop one, and make sure the other session is still shown in the call stack view.

I agree with that. I am saying that starting with only one session could be different.

**Comment by isidorn** [comment]
Ok. I am overflowed for November, thus assigning to December to make it more wordy.

As for the tree: we are not showing the thread if there is only one thread. If that is the case, then this is a bug (which I cannot repro).
If you are suggesting something different than I appologise since I misunderstood you, can you clarify?

**Comment by roblourens** [comment]
I think we agree but I can repro it (debugging with the new DA) and it is a session, not a thread.

**Comment by isidorn** [comment]
Ok. I just tried pwa and yes I can reproduce this. And as you say there are two sessions here the parent session and the child session.

What you are suggesting is the following:
* If there is only one child session, do not render the child session - only render the parent

If you think this is best for the PWA experience than I support it. However there are some other clients of child sessions like cluster debugging and python (I think)
So let's wait and see what is @weinand @qubitron and @connor4312 feedback

**Comment by roblourens** [comment]
I think I am saying only render the child, not only render the parent.

**Comment by qubitron** [comment]
@anapo14 any feedback on this?

**Comment by connor4312** [comment]
Maybe instead of showing "Paused" / "Running" text, we grey out the threads that are running, and then set the title based on the focused stack frame? That gets rid of the duplication and feels intuitive to me:

![image](https://user-images.githubusercontent.com/2230985/69977698-1ec9e880-14e0-11ea-85fd-4b9ddd375b5d.png)

Collapsing when there's only a single child of the debug session would the default PWA case, but it feels like it might be too special-cased towards PWA. Also, if the session is spawning short-lived children regularly, this will cause the call stack to 'flicker' between showing a nested pair of child sessions and the single collapsed child, which may be confusing and cause usability issues.

**Comment by weinand** [comment]
saw the same while testing https://github.com/microsoft/vscode/issues/85984

**Comment by isidorn** [comment]
Please try it out from next weeks insiders and let me know what you think about the current approach.

**Comment by connor4312** [comment]
lgtm

**Comment by isidorn** [comment]
@connor4312 thanks for trying it out, adding verified label.
